src/jit/namedintrinsiclist.h

```diff
@@ -75,8 +75,8 @@ enum HWIntrinsicFlag : unsigned int
     // some SIMD scalar intrinsics need the semantics of copying upper bits from the source operand
     HW_Flag_CopyUpperBits = 0x200,
 
-    // Select base type using argument type
-    HW_Flag_BaseTypeFromArg = 0x400,
+    // Select base type using the first argument type
+    HW_Flag_BaseTypeFromFirstArg = 0x400,
```

Comment:
With this change, all the intrinsics (except MemoryStore) that return a scalar value (e.g., `bool`) should explicitly have flag `HW_Flag_BaseTypeFromFirstArg` or `HW_Flag_BaseTypeFromSecondArg`.

Reply:
Speaking of asserts (which we were a few hours ago on another thread), this seems like something that could be asserted somewhere - e.g. a pass over the table that validates these kinds of properties. There are probably other consistency checks across signature, category and flag that could be checked.

Reply:
@fiigii, could you log an issue tracking this?